pkg/hostagent/hostagent.go

```diff
@@ -259,6 +265,50 @@ func (a *HostAgent) Run(ctx context.Context) error {
 		return err
 	}
 
+	if *a.y.Video.Display == "vnc" {
+		vncdisplay, vncoptions, _ := strings.Cut(*a.y.Video.VNC.Display, ",")
+		vnchost, vncnum, err := net.SplitHostPort(vncdisplay)
+		if err != nil {
+			return err
+		}
+		n, err := strconv.Atoi(vncnum)
+		if err != nil {
+			return err
+		}
+		vncport := strconv.Itoa(5900 + n)
+		vncpwdfile := filepath.Join(a.instDir, filenames.VNCPasswordFile)
+		vncpasswd, err := generatePassword(8)
+		if err != nil {
+			return err
+		}
+		if err := a.driver.ChangeDisplayPassword(ctx, vncpasswd); err != nil {
+			return err
+		}
+		if err := os.WriteFile(vncpwdfile, []byte(vncpasswd), 0600); err != nil {
+			return err
+		}
+		if strings.Contains(vncoptions, "to=") {
+			vncport, err = a.driver.GetDisplayConnection(ctx)
+			if err != nil {
+				return err
+			}
+			p, err := strconv.Atoi(vncport)
+			if err != nil {
+				return err
+			}
+			vncnum = strconv.Itoa(p - 5900)
+			vncdisplay = net.JoinHostPort(vnchost, vncnum)
+		}
+		vncfile := filepath.Join(a.instDir, filenames.VNCDisplayFile)
+		if err := os.WriteFile(vncfile, []byte(vncdisplay), 0600); err != nil {
+			return err
+		}
+		vncurl := "vnc://:" + vncpasswd + "@" + net.JoinHostPort(vnchost, vncport)
+		logrus.Infof("VNC server running at <%s>", vncurl)
+		logrus.Infof("VNC Display: \"%s\" `%s`", vncdisplay, vncfile)
+		logrus.Infof("VNC Password: \"%s\" `%s`", vncpasswd, vncpwdfile)
```

Comment:
The raw password string shouldn't be printed. Just print the password file path would be ok.

Reply:
Then the link won't work and it would be more hassle for the user (running the `cat`), but can do this.

Reply:
Maybe we should still have a `limactl vnc INSTANCE` command after all, if we can't have clickable links (and I reluctantly agree that printing passwords is always bad).

I know I argued against the vnc subcommand before. I still don't think it should embed a VNC client, but just run `open vnc://...` for the user.

Reply:
Maybe call it `viewer` or something, in case there is some other protocol added in the future ?

Reply:
And it's not really a password, it's more like a PIN code or a little like a vnc token (although always granted)

Reply:
You can click the link (like before), just that you also need to retrieve the password from the file...

Reply:
```
INFO[0001] [hostagent] VNC server running at 127.0.0.1:0 <vnc://127.0.0.1:5900> 
INFO[0001] [hostagent] VNC Display: `/home/anders/.lima/default/vncdisplay` 
INFO[0001] [hostagent] VNC Password: `/home/anders/.lima/default/vncpassword` 
```